R/financial_functions.R

```diff
@@ -16,7 +16,37 @@
 #'   under baseline and late and sudden, market prices/costs, company net profit
 #'   margins, the proximity to target in the production forecast period and an
 #'   indication of the direction of the technology.
-calculate_net_profits <- function(data) {
+#' @param shock_year A numeric vector of length one that indicates in which year
+#'   the policy shock strikes in a given scenario.
+#' @param end_year Numeric, holding end year of analysis.
+#' @param carbon_data NGFS carbon prices.
+calculate_net_profits <- function(data,
+                                  shock_year,
+                                  end_year,
+                                  carbon_data) {
+
+  carbon_data <- carbon_data %>% dplyr::filter(.data$model == "MESSAGEix-GLOBIOM 1.0") %>% dplyr::select(-c(scenario_geography))
+  market_passthrough <- 0
+
+  data <-data %>%
+    merge(carbon_data, by = c('year')) %>%
+    dplyr::mutate(
+      carbon_tax = ifelse (year > shock_year, .data$carbon_tax ,0),
+      production_compensation = .data$late_sudden - .data$baseline,
+      net_profits_baseline = .data$baseline * .data$Baseline_price * .data$net_profit_margin,
+      net_profits_ls = dplyr::if_else(
+        .data$direction == "declining",
+        .data$late_sudden * (.data$late_sudden_price - (1-market_passthrough) * .data$carbon_tax* .data$emission_factor) * .data$net_profit_margin,
+        .data$late_sudden * .data$late_sudden_price * .data$net_profit_margin -
+          .data$production_compensation * .data$late_sudden_price * .data$net_profit_margin * (1 - .data$proximity_to_target)
+        # TODO: ADO4109 - should the market size penalty only be applied to laggards?
+
+      )
+    )
+
+}
+
+calculate_net_profits_lrisk <- function(data) {
```

Comment:
You might want to change the naming if you want to be very good. The difference per se between the functions is not that one is for lrisk, but that one calculates net profits with carbon tax and one without.

Reply:
ok makes sense ill adjust it